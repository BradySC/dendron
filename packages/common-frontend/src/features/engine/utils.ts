import _ from "lodash";
import { EngineState } from "./slice";

export const hasInitialized = (engine: EngineState) => {
  return (
    engine.loading === "idle" &&
    !_.isEmpty(engine.notes) &&
    !_.isEmpty(engine.vaults)
  );
};
