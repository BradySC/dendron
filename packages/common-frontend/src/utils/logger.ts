import Amplify, { Logger } from "@aws-amplify/core";

export enum LOG_LEVEL {
  DEBUG = "DEBUG",
  INFO = "INFO",
  ERROR = "ERROR",
}

export function createLogger(name: string) {
  return new Logger(name);
}

export function setLogLevel(lvl: LOG_LEVEL) {
  Amplify.Logger.LOG_LEVEL = lvl;
}
