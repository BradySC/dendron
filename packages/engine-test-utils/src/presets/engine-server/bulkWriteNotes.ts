import { NoteUtils } from "@dendronhq/common-all";
import { TestPresetEntryV4 } from "@dendronhq/common-test-utils";
import _ from "lodash";

const SCHEMAS = {};
const NOTES = {
  BASIC: new TestPresetEntryV4(async ({ vaults, engine }) => {
    const vault = vaults[0];
    const orig = _.size(engine.notes);
    const note1 = NoteUtils.create({
      id: "bar1",
      fname: "bar1",
      created: 1,
      updated: 1,
      vault,
    });
    const note2 = NoteUtils.create({
      id: "bar2",
      fname: "bar2",
      created: 1,
      updated: 1,
      vault,
    });
    const rootNote = (await engine.findNotes({ fname: "root", vault }))[0];
    await engine.bulkWriteNotes({ notes: [note1, note2] });
    return [
      {
        actual: _.size(engine.notes),
        expected: orig + 2,
        msg: "should be 2 more notes",
      },
      {
        expected: engine.notes["bar1"].id,
        actual: note1.id,
      },
      {
        expected: engine.notes["bar1"].fname,
        actual: note1.fname,
      },
      {
        expected: engine.notes["bar1"].vault,
        actual: note1.vault,
      },
      {
        expected: engine.notes["bar1"].parent,
        actual: rootNote.id,
      },
    ];
  }),
};
export const ENGINE_BULK_WRITE_NOTES_PRESETS = {
  NOTES,
  SCHEMAS,
};
