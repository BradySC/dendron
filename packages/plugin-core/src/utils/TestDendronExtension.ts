// import {
//   WorkspaceType,
//   DWorkspaceV2,
//   WorkspaceSettings,
// } from "@dendronhq/common-all";
// import { IWorkspaceService } from "@dendronhq/engine-server";
// import { ExtensionContext, FileSystemWatcher, Disposable } from "vscode";
// import { ICommandFactory } from "../commandFactoryInterface";
// import { ILookupControllerV3Factory } from "../components/lookup/LookupControllerV3Interface";
// import {
//   INoteLookupProviderFactory,
//   ISchemaLookupProviderFactory,
// } from "../components/lookup/LookupProviderV3Interface";
// import { IDendronExtension } from "../dendronExtensionInterface";
// import { IEngineAPIService } from "../services/EngineAPIServiceInterface";
// import { INoteSyncService } from "../services/NoteSyncService";
// import { ISchemaSyncService } from "../services/SchemaSyncServiceInterface";
// import { IWSUtilsV2 } from "../WSUtilsV2Interface";

// export class TestDendronExtension implements IDendronExtension {
//   port?: number | undefined;
//   context: ExtensionContext;
//   serverWatcher?: FileSystemWatcher | undefined;
//   type: WorkspaceType;
//   wsUtils: IWSUtilsV2;
//   commandFactory: ICommandFactory;
//   schemaSyncService: ISchemaSyncService;
//   workspaceService?: IWorkspaceService | undefined;
//   noteSyncService: INoteSyncService;
//   lookupControllerFactory: ILookupControllerV3Factory;
//   noteLookupProviderFactory: INoteLookupProviderFactory;
//   schemaLookupProviderFactory: ISchemaLookupProviderFactory;
//   pauseWatchers<T = void>(cb: () => Promise<T>): Promise<T> {
//     throw new Error("Method not implemented.");
//   }
//   getClientAPIRootUrl(): Promise<string> {
//     throw new Error("Method not implemented.");
//   }
//   getDWorkspace(): DWorkspaceV2 {
//     throw new Error("Method not implemented.");
//   }
//   getWorkspaceImplOrThrow(): DWorkspaceV2 {
//     throw new Error("Method not implemented.");
//   }
//   getWorkspaceSettings(): Promise<WorkspaceSettings | undefined> {
//     throw new Error("Method not implemented.");
//   }
//   getWorkspaceSettingsSync(): WorkspaceSettings | undefined {
//     throw new Error("Method not implemented.");
//   }
//   getDendronWorkspaceSettingsSync():
//     | Partial<{
//         "dendron.dailyJournalDomain": string;
//         "dendron.defaultJournalName": string;
//         "dendron.defaultJournalDateFormat": string;
//         "dendron.defaultJournalAddBehavior": string;
//         "dendron.defaultScratchName": string;
//         "dendron.defaultScratchDateFormat": string;
//         "dendron.defaultScratchAddBehavior": string;
//         "dendron.copyNoteUrlRoot": string;
//         "dendron.linkSelectAutoTitleBehavior": string;
//         "dendron.defaultLookupCreateBehavior": string;
//         "dendron.defaultTimestampDecorationFormat": string;
//         "dendron.rootDir": string;
//         "dendron.dendronDir": string;
//         "dendron.logLevel": string;
//         "dendron.trace.server": string;
//         "dendron.serverPort": string;
//       }>
//     | undefined {
//     throw new Error("Method not implemented.");
//   }
//   getWorkspaceSettingOrDefault({
//     wsConfigKey,
//     dendronConfigKey,
//   }: {
//     wsConfigKey:
//       | "dendron.dailyJournalDomain"
//       | "dendron.defaultJournalName"
//       | "dendron.defaultJournalDateFormat"
//       | "dendron.defaultJournalAddBehavior"
//       | "dendron.defaultScratchName"
//       | "dendron.defaultScratchDateFormat"
//       | "dendron.defaultScratchAddBehavior"
//       | "dendron.copyNoteUrlRoot"
//       | "dendron.linkSelectAutoTitleBehavior"
//       | "dendron.defaultLookupCreateBehavior"
//       | "dendron.defaultTimestampDecorationFormat"
//       | "dendron.rootDir"
//       | "dendron.dendronDir"
//       | "dendron.logLevel"
//       | "dendron.trace.server"
//       | "dendron.serverPort";
//     dendronConfigKey: string;
//   }) {
//     throw new Error("Method not implemented.");
//   }
//   setupViews(context: ExtensionContext): Promise<void> {
//     throw new Error("Method not implemented.");
//   }
//   addDisposable(disposable: Disposable): void {
//     throw new Error("Method not implemented.");
//   }
//   getEngine(): IEngineAPIService {
//     throw new Error("Method not implemented.");
//   }
// }
