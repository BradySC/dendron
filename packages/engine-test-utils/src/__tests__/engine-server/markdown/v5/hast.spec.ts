/**
 * Test HTML specific transformations
 */
import { DendronASTDest, ProcFlavor } from "@dendronhq/engine-server";
import { ENGINE_HOOKS } from "../../../../presets";
import { createProcCompileTests } from "../utils";
import { getOpts, runTestCases } from "./utils";

describe("GIVEN remark", () => {
	describe("WHEN table is present", () => {
		runTestCases(
			createProcCompileTests({
				name: "table is present",
				setup: async (opts) => {
					const { proc } = getOpts(opts);
					const txt = `
| Software Name | Comparison w/ Dendron                        | Flexible Hierarchy | Open source | Local-first | Fast and performant | Bi-directional links | Outlining |
| ------------- | -------------------------------------------- | ------------------ | ----------- | ----------- | ------------------- | -------------------- | --------- |
| Dendron       | How?                                         | ✅                 | ✅          | ✅          | ✅                  | ✅                   | ❌        |`
					const resp = await proc.process(txt);
					return { resp, proc };
				},
				verify: {
					[DendronASTDest.HTML]: {
						[ProcFlavor.REGULAR]: async ({ extra }) => {
							const { resp } = extra;
							expect(resp).toMatchSnapshot();
							// await checkString(resp.contents, "No note with name alpha found");
						},
						[ProcFlavor.PREVIEW]: ProcFlavor.REGULAR,
						[ProcFlavor.PUBLISHING]: ProcFlavor.REGULAR,
					},
				},
				preSetupHook: async (opts) => {
					await ENGINE_HOOKS.setupBasic({ ...opts, extra: { idv2: true } });
				},
			})
		);
	});
});
