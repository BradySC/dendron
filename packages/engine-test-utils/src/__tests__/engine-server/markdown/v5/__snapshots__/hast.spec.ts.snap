// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN remark WHEN table is present "HTML: table is present: PREVIEW" 1`] = `
VFile {
  "contents": "<h1 id=\\"foo\\">Foo</h1>

























<table class=\\"responsive\\"><thead><tr><th>Software Name</th><th>Comparison w/ Dendron</th><th>Flexible Hierarchy</th><th>Open source</th><th>Local-first</th><th>Fast and performant</th><th>Bi-directional links</th><th>Outlining</th></tr></thead><tbody><tr><td>Dendron</td><td>How?</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>❌</td></tr></tbody></table>
<hr>
<strong>Children</strong>
<ol>
<li><a href=\\"foo.ch1\\">Ch1</a></li>
</ol>",
  "cwd": "<PROJECT_ROOT>",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`GIVEN remark WHEN table is present "HTML: table is present: PUBLISHING" 1`] = `
VFile {
  "contents": "<h1 id=\\"foo\\"><a aria-hidden=\\"true\\" class=\\"anchor-heading icon-link\\" href=\\"#foo\\"></a>Foo</h1>

























<table class=\\"responsive\\"><thead><tr><th>Software Name</th><th>Comparison w/ Dendron</th><th>Flexible Hierarchy</th><th>Open source</th><th>Local-first</th><th>Fast and performant</th><th>Bi-directional links</th><th>Outlining</th></tr></thead><tbody><tr><td>Dendron</td><td>How?</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>❌</td></tr></tbody></table>
<hr>
<strong>Children</strong>
<ol>
<li><a href=\\"/notes/foo.ch1\\">Ch1</a></li>
</ol>",
  "cwd": "<PROJECT_ROOT>",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`GIVEN remark WHEN table is present "HTML: table is present: REGULAR" 1`] = `
VFile {
  "contents": "<h1 id=\\"foo\\">Foo</h1>

























<table class=\\"responsive\\"><thead><tr><th>Software Name</th><th>Comparison w/ Dendron</th><th>Flexible Hierarchy</th><th>Open source</th><th>Local-first</th><th>Fast and performant</th><th>Bi-directional links</th><th>Outlining</th></tr></thead><tbody><tr><td>Dendron</td><td>How?</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>✅</td><td>❌</td></tr></tbody></table>
<hr>
<strong>Children</strong>
<ol>
<li><a href=\\"foo.ch1\\">Ch1</a></li>
</ol>",
  "cwd": "<PROJECT_ROOT>",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;
